libs/api_common/windows_platform_common.cpp

```diff
@@ -29,59 +29,118 @@
 // the preprocessor treat a prefix list as one macro argument.
 #define COMMA ,
 
+//
+// XDP program type.
+//
 const ebpf_context_descriptor_t g_xdp_context_descriptor = {
     sizeof(xdp_md_t),
     EBPF_OFFSET_OF(xdp_md_t, data),
     EBPF_OFFSET_OF(xdp_md_t, data_end),
     EBPF_OFFSET_OF(xdp_md_t, data_meta)};
 
-const ebpf_windows_program_type_data_t windows_xdp_program_type_data = {EBPF_PROGRAM_TYPE_XDP, EBPF_ATTACH_TYPE_XDP};
-
 const EbpfProgramType windows_xdp_program_type =
-    PTYPE("xdp", &g_xdp_context_descriptor, (uint64_t)&windows_xdp_program_type_data, {"xdp"});
-
-const ebpf_windows_program_type_data_t windows_bind_program_type_data = {EBPF_PROGRAM_TYPE_BIND, EBPF_ATTACH_TYPE_BIND};
+    PTYPE("xdp", &g_xdp_context_descriptor, (uint64_t)&EBPF_PROGRAM_TYPE_XDP, {"xdp"});
 
+//
+// Bind program type.
+//
 const ebpf_context_descriptor_t g_bind_context_descriptor = {
     sizeof(bind_md_t), EBPF_OFFSET_OF(bind_md_t, app_id_start), EBPF_OFFSET_OF(bind_md_t, app_id_end), -1};
 
 const EbpfProgramType windows_bind_program_type =
-    PTYPE("bind", &g_bind_context_descriptor, (uint64_t)&windows_bind_program_type_data, {"bind"});
+    PTYPE("bind", &g_bind_context_descriptor, (uint64_t)&EBPF_PROGRAM_TYPE_BIND, {"bind"});
 
+//
+// eBPF Sample extensions program type.
+//
 const ebpf_context_descriptor_t g_sample_ext_context_descriptor = {
     sizeof(sample_program_context_t),
     EBPF_OFFSET_OF(sample_program_context_t, data_start),
     EBPF_OFFSET_OF(sample_program_context_t, data_end),
     -1, // Offset into ctx struct for pointer to metadata, or -1 if none.
 };
 
-const ebpf_windows_program_type_data_t windows_sample_program_type_data = {
-    EBPF_PROGRAM_TYPE_SAMPLE, EBPF_ATTACH_TYPE_SAMPLE};
-
 const EbpfProgramType windows_sample_ext_program_type =
-    PTYPE("sample_ext", &g_sample_ext_context_descriptor, (uint64_t)&windows_sample_program_type_data, {"sample_ext"});
+    PTYPE("sample_ext", &g_sample_ext_context_descriptor, (uint64_t)&EBPF_PROGRAM_TYPE_SAMPLE, {"sample_ext"});
+
+//
+// CGROUP_SOCK_ADDR.
+//
+const ebpf_context_descriptor_t g_sock_addr_context_descriptor = {
+    sizeof(bpf_sock_addr_t),
+    -1, // Offset into ctx struct for pointer to data, or -1 if none.
+    -1, // Offset into ctx struct for pointer to data, or -1 if none.
+    -1, // Offset into ctx struct for pointer to metadata, or -1 if none.
+};
```

Comment:
This looks like it duplicates what is in [net_ebpf_ext_program_info.h](https://github.com/microsoft/ebpf-for-windows/pull/858/files#diff-f813ac74105e8bf407c730c9ab3ac36211f601d2cc9ffabfd46058741d062a2a) line 36.
Can we have it in a common header instead of duplicating it?